seahorn/lib/klee_string_helper.c

```diff
@@ -2,38 +2,9 @@
 #include <string_helper.h>
 #include <seahorn/seahorn.h>
 
-extern void __assert_fail (const char *__assertion, const char *__file,
-			   unsigned int __line, const char *__function);
-# define klee_assert(expr)                                            \
-  ((expr)                                                             \
-   ? (void) (0)                                                       \
-   : __assert_fail (#expr, __FILE__, __LINE__, __PRETTY_FUNCTION__))  \
-
-size_t dispatch_alloc_size(size_t sz) {
-  switch(sz) {
-    case 1: return 1;
-    case 2: return 2;
-    case 3: return 3;
-    case 4: return 4;
-    case 5: return 5;
-    case 6: return 6;
-    case 7: return 7;
-    case 8: return 8;
-    case 9: return 9;
-    case 10: return 10;
-    case 11: return 11;
-    case 12: return 12;
-    case 13: return 13;
-    case 14: return 14;
-    case 15: return 15;
-    case 16: return 16;
-    default: return sz <= 4096 ? 4096 : sz; 
-  }
-}
-
 struct aws_string *ensure_string_is_allocated(size_t len) {
   size_t alloc_size = sizeof(struct aws_string) + len + 1;
-  struct aws_string *str = malloc(dispatch_alloc_size(alloc_size));
+  struct aws_string *str = bounded_malloc(alloc_size);
```

Comment:
I thought symbolic allocations are not allowed. This seem to be turned into a symbolic allocation.

Reply:
But the variable `alloc_size` depends on symbolic value `len`, we have a constraint for `alloc_size` in terms of `len`. When we do the allocation, klee requires a concretize size.

Reply:
which is why I thought that `alloc_size` has to be wrapped in `dispatch_alloc_size`